a10_neutron_lbaas/v2/handler_listener.py

```diff
@@ -260,21 +261,22 @@ def update(self, context, old_listener, listener):
         with a10.A10WriteStatusContext(self, context, listener) as c:
             self._update(c, context, listener)
 
-    def _delete_listener(self, c, context, listener):
+    def _delete_listener(self, c, context, listener, protocol):
```

Comment:
... `_delete_listener` takes an acos protocol.

You're going to need to explain to me why always deleting it by what's on the listener isn't correct. Is it because the protocol is used as part of the key, and the name for the vport doesn't get used?